packages/json/src/lib/JSONProvider.ts

```diff
@@ -550,60 +550,70 @@ export class JSONProvider<StoredValue = unknown> extends JoshProvider<StoredValu
   }
 
   public async [Method.Random](payload: Payload.Random<StoredValue>): Promise<Payload.Random<StoredValue>> {
-    const { count, duplicates } = payload;
+    const { count, unique } = payload;
     const size = await this.handler.size();
 
-    if (size === 0) return { ...payload, data: [] };
-    if (size < count) {
+    if (unique && size < count) {
       payload.errors.push(this.error({ identifier: CommonIdentifiers.InvalidCount, method: Method.Random }, { size }));
 
       return payload;
     }
 
+    if (size === 0) {
+      payload.errors.push(this.error({ identifier: CommonIdentifiers.MissingData, method: Method.Random }, { unique, count }));
+
+      return payload;
+    }
+
     payload.data = [];
 
     const keys = await this.handler.keys();
 
-    if (duplicates) {
-      while (payload.data.length < count) {
-        const key = keys[Math.floor(Math.random() * size)];
-
-        payload.data.push((await this.handler.get(key))!);
-      }
-    } else {
+    if (unique) {
       const randomKeys = new Set<string>();
 
       while (randomKeys.size < count) randomKeys.add(keys[Math.floor(Math.random() * keys.length)]);
```

Comment:
Are we set on using this randomness, it's not the *most* random

Reply:
Yeah. The random method isn't like storing your credit card info so it doesn't need to be fully random, pseudorandom is sufficient